### PR TITLE
Add Fake.DotNet.Testing.Coverlet

### PR DESCRIPTION
**Pull request by Tarmil**
### Description

Add a module to run [Coverlet](https://github.com/tonerdo/coverlet) test coverage as part of `dotnet test`.

## TODO

- [x] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [x] unit tests
- [ ] integration test
- [x] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)     
- [x] (if new module) the module has been linked from the "Modules" menu, edit `help/templates/template.cshtml`, linking to the API-reference is fine.
- [x] (if new module) the module is in the correct namespace
- [x] (if new module) the module is added to Fake.sln (`dotnet sln Fake.sln add src/app/Fake.*/Fake.*.fsproj`)
- [x] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design-Guidelines) is honored

### CODE REVIEW

File: src/app/Fake.DotNet.Testing.Coverlet/Coverlet.fs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        Threshold : int option
+        /// Type of coverage to check against the threshold.
+        ThresholdType : ThresholdType
+        /// Coverage statistic to check against the threshold.
+        ThresholdStat : ThresholdStat
+        /// Generate results with URL links from SourceLink instead of file paths.
+        UseSourceLink : bool }
+
+/// The default parameters.
+let Defaults =
```
Comment (matthid) [inline src/app/Fake.DotNet.Testing.Coverlet/Coverlet.fs]: Can we make that `internal`?

File: src/app/Fake.DotNet.Testing.Coverlet/Coverlet.fs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    | ThresholdType.Branch -> "branch"
+    | ThresholdType.Method -> "method"
+
+let private thresholdStatToString = function
+    | ThresholdStat.Minimum -> "minimum"
+    | ThresholdStat.Total -> "total"
+    | ThresholdStat.Average -> "average"
+
+/// Add Coverlet parameters to the MSBuild command.
+let withMSBuildArguments (param: CoverletParams -> CoverletParams) (args: MSBuild.CliArguments) =
```
Comment (matthid) [inline src/app/Fake.DotNet.Testing.Coverlet/Coverlet.fs]: Interesting approach, currently we don't have any module that works like this. Is there any other API design you have considered (I mean it definitely looks reasonable to me, just asking)

In any case, I think we need inline docs for this as well.

**Comment by matthid** [comment]
Also, feel free to merge release/next to get the build green.

**Comment by Tarmil** [comment]
> One other thing: There is also a standalone tool so should we make sure the `CoverletParams` type is compatible with it as well? So we can add additional functions later which run the global tool? Or should we just add it later (and a second type if needed)

Good point. The standalone tool seems to work similarly to DotCover so I'll see if I can add a module with a similar API to Fake.DotNet.Testing.DotCover.

**Comment by matthid** [comment]
Note that I don't expect an implementation tight away - though that would be an option as well. I was just pointing out if this is something to consider from an API and naming perspective

**Comment by matthid** [comment]
Thanks again!

**Pull request merged**